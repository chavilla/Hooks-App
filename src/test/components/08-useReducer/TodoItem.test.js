const { shallow } = require("enzyme");
const {
  default: TodoItem,
} = require("../../../components/08-useReducer/TodoItem");
const { state } = require("../../fixtures/toDoList");

describe("TodoItem testing", () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoItem
      todo={state}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
      i={0}
    />
  );

  // the testings
  test("should ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("verify that the funtions delete have been called", () => {

    const btn=wrapper.find('.btn-danger');

    btn.simulate('click');

    expect(handleDelete).toHaveBeenCalledWith(state.id);
  });

  test("verify that the funtions toggle have been called", () => {
    
    wrapper.find('p').simulate('click');

    expect(handleToggle).toHaveBeenCalledWith(state.id);
  });

  test('should to show the information', () => {
    
    expect(wrapper.find('p').text()).toBe(`1-${state.desc}`);

  });

  test('should to have the classname complete if the todo.done true', () => {
    expect(wrapper.find('p').hasClass('complete')).toBe(true)
  })
  
  

});
