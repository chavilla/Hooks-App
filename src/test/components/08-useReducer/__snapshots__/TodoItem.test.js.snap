// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem testing should  1`] = `
<li
  className="d-flex row my-2 w-100 align-items-center justify-content-around"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    -
  </p>
  <button
    className="btn btn-danger col-4"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</li>
`;
